### PR TITLE
[feature](table-valued-function) Support `desc from s3()` and modify the syntax of tvf

### LINKED ISSUE #10217: [Feature] support table valued function.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

A table-valued function (TVF) is a function that returns a relation or a set of rows. There are two types of TVFs :
- a TVF that can be specified in a FROM clause, e.g. range;
- a TVF that can be specified in SELECT/LATERAL VIEW clauses, e.g. explode.

Just like the definition of spark and clickhouse：
- spark: https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-tvf.html.
- clickhouse: https://clickhouse.com/docs/en/sql-reference/table-functions.  Currently Clickhouse only support from clause table function

The table valued function can be used to test our vector engines performance.

### Use case

The table valued function works like this：
MySQL [(none)]> select sum(number) from (select number from numbers("1000") where number < 10) as c;
+---------------+
| sum(number) |
+---------------+
| 45 |
+---------------+

That is: numbers("1000") returns a temporary table which only has one column named 'number' and it's row-values are from 0 to 1000.

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #10217

This pr does two things:
1. Support `desc function s3()`
2. modify the syntax of tvf

## Problem summary

1. Support `desc function s3()`
Use `desc function s3()` to get table schema of S3 table valued function. It likes:
```
MySQL [(none)]> desc function s3("URI" = "http://127.0.0.1:9312/test2/student_with_names.csv","ACCESS_KEY"= "minioadmin", "SECRET_KEY" = "minioadmin", "FORMAT" = "csv_with_names");
+-------+------+------+-------+---------+-------+
| Field | Type | Null | Key   | Default | Extra |
+-------+------+------+-------+---------+-------+
| id    | TEXT | Yes  | false | NULL    | NONE  |
| name  | TEXT | Yes  | false | NULL    | NONE  |
| age   | TEXT | Yes  | false | NULL    | NONE  |
+-------+------+------+-------+---------+-------+
3 rows in set (0.32 sec)
```
2. modify the syntax of tvf
In the old syntax, the params of tvf is fixed order and is not flexible enough for S3 tvf. So, We change the syntax of table valued function. 

For numbers tvf:
```
// old syntax:
numbers("n", "m")

// new syntax:
numbers(
  "number" = "n",
  "backend_num" = "m"
  );
```
For S3 tvf:
```
// old syntax:
s3("http://127.0.0.1:9312/test2/student_with_names.csv", "minioadmin",  "minioadmin", "csv_with_names");

// new syntax:
s3(
"URI" = "http://127.0.0.1:9312/test2/student_with_names.csv",
ACCESS_KEY"= "minioadmin", 
"SECRET_KEY" = "minioadmin", 
"FORMAT" = "csv_with_names"
);
```

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
5. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.19 seconds
 load time: 458 seconds
 storage size: 17180198768 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221107190050_clickbench_pr_41419.html

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -3813,6 +3813,10 @@ describe_stmt ::=
     {:
         RESULT = new DescribeStmt(table, false);
     :}
+    | describe_command KW_FROM table_valued_function_ref:tvf
```
[the remark this answers is not in the document]
Comment (BePPPower) [reply fe/fe-core/src/main/cup/sql_parser.cup]: ok

**Review by morningman — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**